extension/lib/bigcommerce/AuthRepository.js

```diff
@@ -0,0 +1,42 @@
+const jwt = require('jwt-simple')
+const crypto = require('crypto')
+
+/**
+ * @property {string} _clientId
+ * @property {string} _storeHash
+ * @property {string} _clientSecret
+ */
+class AuthRepository {
+  /**
+   * @param {string} clientId
+   * @param {string} storeHash
+   * @param {string} clientSecret
+   */
+  constructor (clientId, storeHash, clientSecret) {
+    this._clientId = clientId
+    this._storeHash = storeHash
+    this._clientSecret = clientSecret
+  }
+
+  /**
+   * @param {string} customerId
+   * @param {string|null} redirectLink
+   */
+  preAuthToken (customerId, redirectLink = null) {
```

Comment:
Do you see a reason for redirect url to be optional? I know we have this in legacy, but here there is only one usage and the link should always be provided.

Reply:
I agree. I made a change to make the redirect link to be required.